lib/internal/bootstrap/loaders.js

```diff
@@ -208,6 +211,73 @@
         NativeModule.require;
       fn(this.exports, requireFn, this, process);
 
+      if (config.experimentalModules) {
+        if (this.id === 'events') {
+          this.namespace = [];
+        } else {
+          this.namespace = Object.entries(
+            Object.getOwnPropertyDescriptors(this.exports))
+            .filter(([name, d]) => d.enumerable)
+            .map(([name]) => name);
+        }
+
+        const update = (property, value) => {
+          if (this.reflect !== undefined && this.namespace.includes(property))
+            this.reflect.exports[property].set(value);
+        };
+
+        const methodWrapMap = new WeakMap();
+
+        const proxy = new Proxy(this.exports, {
+          set(target, prop, value) {
+            if (Reflect.set(target, prop, value)) {
+              update(prop, value);
+              return true;
+            }
+            return false;
+          },
+          defineProperty(target, prop, descriptor) {
+            if (Reflect.defineProperty(target, prop, descriptor)) {
+              update(prop, Reflect.get(target, prop, target));
+              return true;
+            }
+            return false;
+          },
+          deleteProperty(target, prop) {
+            if (Reflect.deleteProperty(target, prop)) {
+              update(prop, undefined);
+              return true;
+            }
+            return false;
+          },
+          get(target, prop) {
+            const value = Reflect.get(target, prop, target);
+            if (methodWrapMap.has(value))
+              return methodWrapMap.get(value);
+            if (typeof value === 'function') {
+              function wrap(...args) {
+                if (new.target)
+                  return Reflect.construct(value, args, new.target);
+                const thisArg = this === proxy ? target : this;
+                return Reflect.apply(value, thisArg, args);
+              }
+              const { name, length } = value;
+              Object.defineProperties(wrap, {
+                name: { value: name },
+                length: { value: length },
+              });
+              Object.setPrototypeOf(wrap, value);
+              wrap.prototype = value.prototype;
+              methodWrapMap.set(value, wrap);
+              return wrap;
+            }
+            return value;
+          },
+        });
```

Comment:
Is there a reason why you would prefer that?

Reply:
Yes. I first avoid wrapping for the 99% case (plain or bound functions) and then only wrap with a proxy for the native case. Beyond the fact that less wrapping is good in general this is important  to the `esm` loader because folks can opt for CJS named export support beyond buitlins. Avoiding the the function/proxy wrap means more methods are `===` to other references a user might store. For the 1% case (native method) wrapping with a proxy lets all other values passthru to the original function while I just hook the `apply` trap.

Reply:
I'll have to re-audit our exports to make sure we only need to bind for native methods, but that does seem like a better case

Reply:
@jdalton your proxy unfortunately breaks in certain cases which i think could be a v8 bug...

Reply:
> proxy unfortunately breaks in certain cases which i think could be a v8 bug so i'll keep using my wrap function for now

Can you show me the syntax you were trying to execute?
For example:
```
import { EventEmitter } from "events"
new EventEmitter
```

Reply:
FWIW I've been using the proxy technique for a few months and it seems to work when compiled directly into Node master as well so I'd dig a test case if you've hit an issue for sure.

Reply:
@jdalton i wasn'r running any esm...

Reply:
Dev error? Proxies should work.

Reply:
Can you show how you attempted proxies. Might be able to spot the issue.

Reply:
When `EventEmitter.call({})` is called the `thisArg === proxy` condition is met because `EventEmitter` is the `module.exports` proxy so the `thisArg` is set to the original `EventEmitter` target _(the unwrapped `module.exports)_. This results in a successful invocation.

Reply:
@devsnek Can you find an isolated reproduction of the bug? The following seems to be working just fine here:

```
$ node
> const { EventEmitter } = events
undefined
> Reflect.apply(EventEmitter.call, EventEmitter, [])
TypeError: Cannot set property 'domain' of undefined
    at EventEmitter.init (domain.js:401:15)
    at EventEmitter (events.js:27:21)
```

Reply:
@jdalton i copied yours in, except for using regular proxies instead of your wrapper, which i doubt makes any difference in this case

Reply:
ok i just came back to this and it seems like the proxy works now so i'm going to just assume i need more sleep... i'll push in a few minutes after some more testing